src/lib/rules/http-compression/http-compression.ts

```diff
@@ -0,0 +1,672 @@
+             *    On thing that Zopfli does is that it sets FLG and
+             *    MTIME to 0, XFL to 2, and OS to 3 [1], so basically
+             *    files compressed with Zopfli will most likely start
+             *    with `1f8b 0800 0000 0000 0203`, unless things are
+             *    changed by the user (which in general doesn't seem
+             *    very likely to happen).
+             *
+             *    Now, regular gzip output might also start with that,
+             *    even thought the chance of doing so is smaller:
+             *
+             *    * Most web servers (e.g.: Apache², NGINX³), by default,
+             *      will not opt users into the best compression level,
+             *      therefore, the output shouldn't have XFL set to 2.
+             *
+             *    * Most utilities that output regular gzip will have
+             *      non-zero values for MTIME and FLG.
+             *
+             *   So, if a file does not start with:
+             *
+             *      `1f8b 0800 0000 0000 0203`
+             *
+             *   it's a good (not perfect) indication that Zopfli wasn't
+             *   used, but it's a fast check compared to compressing
+             *   files and comparing file sizes. However, if a file does
+             *   start with that, it can be either Zopfli or gzip, and
+             *   we cannot really make assumptions here.
+             *
+             * - - - - - - - - - - - - - - - - - - - - - - - - - - - - -
+             *
+             * Ref:
+             *
+             *   ¹ https://github.com/google/zopfli/blob/6818a0859063b946094fb6f94732836404a0d89a/src/zopfli/gzip_container.c#L90-L101)
+             *   ² https://httpd.apache.org/docs/current/mod/mod_deflate.html#DeflateCompressionLevel
+             *   ³ https://nginx.org/en/docs/http/ngx_http_gzip_module.html#gzip_comp_level
+             */
+
+            return !checkIfBytesMatch(rawResponse, [0x1f, 0x8b, 0x08, 0x00, 0x00, 0x00, 0x00, 0x00, 0x02, 0x03]);
+        };
+
+        const checkBrotli = async (resource, element) => {
+            const { contentEncodingHeaderValue, rawResponse, response } = await getNetworkData(resource, { 'Accept-Encoding': 'br' });
+
+            const compressedWithBrotli = isCompressedWithBrotli(rawResponse);
+
+            // - - - - - - - - - - - - - - - - - - - - - - - - - - - - -
+
+            // Check if compressed with Brotli over HTTP.
+
+            if (isHTTP(resource)) {
+                if (compressedWithBrotli) {
+                    await context.report(resource, element, generateCompressionMessage('Brotli', true, 'over HTTP'));
+                }
+
+                return;
+            }
+
+            // - - - - - - - - - - - - - - - - - - - - - - - - - - - - -
+
+            // Check compressed vs. uncompressed sizes.
+
+            /*
+             * TODO: Remove the following once connectors
+             *       support Brotli compression.
```

Comment:
Can you open an issue for this?
Basically what we need is `requester` to auto decompress `Brotli`.

Reply:
> Can you open an issue for this?

[Done](https://github.com/sonarwhal/sonarwhal/issues/735).